src/lightning_app/api/http_methods.py

```diff
@@ -29,7 +29,12 @@ def __init__(self, route: str, method: Callable, method_name: Optional[str] = No
             timeout: The time in seconds taken before raising a timeout exception.
         """
         self.route = route
-        self.component_name = method.__self__.name
+        self.is_function = not hasattr(method, "__self__")
```

Comment:
this is unrelated to the PR right?

Reply:
This is required to enable function to be used instead of method of the flow.